app/models/course/announcement_settings.rb

```diff
@@ -3,13 +3,28 @@ class Course::AnnouncementSettings
   include ActiveModel::Conversion
   include ActiveModel::Validations
 
+  validates :title, presence: true
   validates :pagination, numericality: { greater_than: 0 }
 
   # Initialises the settings adapter
   #
-  # @param [#settings] settings The settings object provided by the settings_on_rails gem.
-  def initialize(settings)
-    @settings = settings
+  # @param [#settings] course_settings The settings object provided by the settings_on_rails gem.
+  def initialize(course_settings)
+    @settings = course_settings.settings(:announcement)
+  end
+
+  # Returns the title of announcements component
+  #
+  # @return [String] The custom or default title of announcements component
+  def title
+    @settings.title || I18n.t('course.announcements.sidebar_title')
```

Comment:
How do I differentiate between a user-provided string and the default string? And also the locale?